dlang/psi-impl/src/main/kotlin/io/github/intellij/dlanguage/parser/DLangParser.kt

```diff
@@ -3544,6 +3543,8 @@ internal class DLangParser(private val builder: PsiBuilder) {
                 bookmark.drop()
                 if (currentIs(DlangTypes.OP_DOT)) {
                     advance()
+                    m.done(DlangTypes.QUALIFIED_IDENTIFIER)
+                    m = m.precede()
```

Comment:
The core of the fix that make Foo!Bar.x being parsed as `QualifiedIdentifier(x, QualifiedIdentifier(Foo!Bar.))` instead of `QualifiedIdentifier(Foo!Bar.x)`.